tests/test_options.py

```diff
@@ -170,38 +172,120 @@ def cli(message):
 
 
 @pytest.mark.parametrize(
-    ("multiple", "nargs", "default"),
+    ("args", "multiple", "nargs", "default", "expected"),
     [
-        (True, 1, []),
-        (True, 1, [1]),
-        # (False, -1, []),
-        # (False, -1, [1]),
-        (False, 2, [1, 2]),
-        # (True, -1, [[]]),
-        # (True, -1, []),
-        # (True, -1, [[1]]),
-        (True, 2, []),
-        (True, 2, [[1, 2]]),
+        # If multiple values are allowed, defaults should be iterable.
+        ([], True, 1, [], ()),
+        ([], True, 1, (), ()),
+        ([], True, 1, tuple(), ()),
+        ([], True, 1, set(), ()),
+        ([], True, 1, frozenset(), ()),
+        ([], True, 1, {}, ()),
+        # Special values.
+        ([], True, 1, None, ()),
+        ([], True, 1, UNSET, ()),
+        # Number of values are kept as-is in the default.
+        ([], True, 1, [1], (1,)),
+        ([], True, 1, [1, 2], (1, 2)),
+        ([], True, 1, [1, 2, 3], (1, 2, 3)),
+        ([], True, 1, [1.1, 2.2], (1.1, 2.2)),
+        ([], True, 1, ["1", "2"], ("1", "2")),
+        ([], True, 1, [None, None], (None, None)),
+        # XXX Why items of the defaults are converted to strings and not left as
```

Comment:
Is this something you figured out? Should we open another issue?

Reply:
There's a few other places with `# XXX` notes, are these things that still need to be addressed?